### ISSUE TITLE
Chunk cutoffs/borders when decreasing frequency

### ISSUE THREAD

**Issue by Lorelando** [PersonTheCat/CaveGenerator#18]
Cavegenerator 0.14
Forge 14.23.5.2814

When decreasing the frequency of caverns without decreasing the scaling you get wonderfully large and somewhat rare caverns, however with not so wonderful chunk "borders"

```
  caverns: {
    enabled: true
    minHeight: 6
    noise3D: {
      frequency: 0.005
      scale: 0.4
      scaleY: 0.3
    }
  }
```
![2019-03-14_11 03 02](https://user-images.githubusercontent.com/19599163/54348795-11b35380-464a-11e9-9d5e-46f3e5cdebe3.png)

This also happen with relatively small scalings:
`      frequency: 0.005
      scale: 0.18
      scaleY: 0.27`
![2019-03-14_11 22 06](https://user-images.githubusercontent.com/19599163/54349453-93f04780-464b-11e9-8fd7-b40192478f30.png)

Activating the strict presets config option will also not crash the game even with those generation errors occurring.

**Comment by PersonTheCat** [comment]
Huh. This looks like a bug I fixed in 0.14. Are you using the latest version from last night?

**Comment by Lorelando** [comment]
Yep, most definitely. Also just re-downloaded it and tried again to make absolutely sure I'm not using an earlier 0.14 build.

**Comment by PersonTheCat** [comment]
Oh okay, thanks. I'll see if I can reproduce it when I get home from work today.

**Comment by PersonTheCat** [comment]
I did a quick fly by and didn't see anything like that. Any way you could send me a seed and coordinates where this issue occurs? I'm assuming this is just the underground forest preset with the specified parameters changed and nothing else.

**Comment by Lorelando** [comment]
Whole preset with only frequency changed:
`
    noise3D: {
      frequency: 0.01
      scale: 0.4
      scaleY: 0.3
    }
`
https://gist.github.com/Lorelando/40c5980d3390082caae615490beac8a9
Seed: -6149155176181929120
X/Y/Z: 260 / 25 / 283
![2019-03-16_19 39 00](https://user-images.githubusercontent.com/19599163/54480003-b5a42700-4823-11e9-8803-0cbbeb811d58.png)
Forge 14.23.5.2814, no other mods installed besides Cave Generator.
I am running on Java 1.8.0_201, but already tried switching to the Java version that comes with Minecraft and did not get any different generation results.


Default preset with `frequency: 0,167`
Seed: -1869638741433744583
X/Y/Z: -203 / 21 / -7
![2019-03-16_19 47 26](https://user-images.githubusercontent.com/19599163/54480078-9fe33180-4824-11e9-9493-b27d6f1ddc70.png)
MD5 (Using "certutil -hashfile CaveGenerator-0.14.jar md5" through Windows CMD): 1e309fc36de6d32e5208899c79203950

**Comment by PersonTheCat** [comment]
Okay, I can confirm that this problem does occur in MC 1.12.2, but not in 1.12.1. That seems to have been the difference. This might be a nightmare to fix. :/
![2019-03-17_12 48 26](https://user-images.githubusercontent.com/33143148/54495616-25adbe00-48b3-11e9-9dfd-ab9898475fc9.png)

**Comment by PersonTheCat** [comment]
Got it. This was the same seed change bug I was referring to. The problem was related to what I assume are optimizations between versions which change the way that worlds are unloaded. You can try the new version below to verify that everything is working, or wait until changes are uploaded here and compile it from scratch. Thanks for your help!
[CaveGenerator-0.15.zip](https://github.com/PersonTheCat/CaveGenerator/files/2975521/CaveGenerator-0.15.zip)

**Comment by Lorelando** [comment]
Sadly the same bug still happens:
![2019-03-18_06 32 35](https://user-images.githubusercontent.com/19599163/54508856-b7670b00-4947-11e9-89b5-8e3ab8b24659.png)

**Comment by PersonTheCat** [comment]
Is this happening anywhere else in the world? If it's just one or two chunks at spawn, this is actually an unrelated bug I've known about since the rewrite. And it isn't related to noise, but the entire generator. What's happening is that I've pushed instantiation to a later event because the World object sent into MabGenCaves is sometimes null when handled by custom world generators. This avoids crashes and was a bit of a trade-off that I knowingly made. I'm still not sure how I want to fix it. ::/

**Comment by Lorelando** [comment]
Huh, I hadn't noticed that, tbh. I stopped looking after finding the glitched chunks at spawn.
I haven't been able to find any other occurrences, so I guess it's fixed :)

**Comment by CREEATION** [comment]
MC: `1.12.2`
Forge: `14.23.5.2824`
Seed (for both tests): `5768783229991403274`
World Type: `Default`

This is still an issue _(not near spawn)_, both with version `0.14` (no other mods other than JourneyMap):
Coords: `/tp Notch -1290 35 852`
![2019-04-11_07 03 43](https://user-images.githubusercontent.com/3277769/55932418-df394e00-5c29-11e9-8870-bf187399ddd2.png)


and `0.15`:
Coords: `/tp Notch -1171 26 827`
![2019-04-11_07 09 03](https://user-images.githubusercontent.com/3277769/55932178-d431ee00-5c28-11e9-90d6-e67524f34e55.png)

My config (`underground_forest.cave`):
```css
  caverns: {
    enabled: true
    minHeight: 16
    maxHeight: 50
    noise3D: {
      frequency: 0.011
      scale: 0.15
      scaleY: 0.3
    }
  }
```

I have some other presets enabled and modified, namely `flooded_vanilla.cave` and `stone_clusters.cave`, but I'd guess that's unrelated? Haven't touched `stone_clusters.cave`, but here's the changes to `flooded_vanilla.cave`:

```css
  caveBlocks: [
    {
      states: water
      minHeight: 14
      maxHeight: 38
      frequency: 0.7
    }
    /* ... [nothing else changed here] */
  ]
  wallDecorators: [
    {
      states: gravel
      chance: 0.15
      minHeight: 0
      maxHeight: 55
      /* ... [nothing else changed here] */
    }
  ]
```

**Comment by PersonTheCat** [comment]
I replicated this setup pretty closely, but did not reproduce the error. Please try the current build I'm using and verify that it's still occurring. Thanks for your help!
[CaveGenerator-0.15.zip](https://github.com/PersonTheCat/CaveGenerator/files/3071218/CaveGenerator-0.15.zip)

Edit: The one thing I didn't try was using all of these presets simultaneously. If it still does occur, try only using underground_forest.cave. If that fixes it, it's most likely a bug related to HeightMapLocator, which I probably wouldn't fix because I have some design changes in mind that would make it irrelevant.

**Comment by CREEATION** [comment]
I've tried your new `0.15` version both with all presets and only the underground forest preset.
Seed as above (`5768783229991403274`)
Coords as above (`/tp Notch -1171 26 827`)

Same issue:

With all three presets:
![2019-04-12_03 59 13](https://user-images.githubusercontent.com/3277769/56007138-fb93c400-5cd7-11e9-949f-2baf2f8df669.png)

Only underground forest:
![2019-04-12_04 02 14](https://user-images.githubusercontent.com/3277769/56007146-0189a500-5cd8-11e9-9286-de3bd4e61dca.png)

Though, I forgot that I changed the `ravines` object in `vanilla.cave`:
```css
  ravines: {
    [...]
    maxHeight: 40
    [...]
  }
```

//e: Alright, still happens with the default values in `vanilla.cave`:
![2019-04-12_04 10 45](https://user-images.githubusercontent.com/3277769/56007439-03079d00-5cd9-11e9-85c8-f8eb2ef165cd.png)

**Comment by PersonTheCat** [comment]
I've applied the settings you said were changed above and recreated the screenshot at the exact coordinates. The md5 hash for the current version of cave generator is `e5562317370b4e72193f43b1c5fd3fad`. I'm currently out of ideas. I may need you to send me the exact preset files you're using in full so I can test them out.
![2019-04-14_14 00 38](https://user-images.githubusercontent.com/33143148/56097742-132ca180-5ebe-11e9-9851-05cc093c97fc.png)

Edit: Also, what's the other mod you're using? Is it just Optifine? If so, which version?
